### DISCUSSION TITLE
Slow query speed using 2.3.5 Milvus

### DISCUSSION THREAD

**Discussion by saratkirana** [milvus-io/milvus#30583] [Q&A and General discussion]
Hello,

I have a 500M records (128 dim) collection loaded into memory with 30 Query Nodes in AWS, Cluster.

A 100 batch search query with topk 50  is taking around 3 sec. I want to improve this, so I tried increasing dataCoord.segment.maxSize to 2048 using helm uprage command and restarted datacoord (kubectl rollout command). But If I go to Attu, Segment_max_size is still set to 512 and no change in number of segments. Can you help me how to update this configuration and any other recommendations to improve this ?

**Comment by yhmo** [comment]
(python sdk)

1. Set "Eventually" consistency level for search() interface, to reduce search latency.
```
collection.search(
  ...
  consistency_level="Eventually",
)
```

2. Try ` collection.compact()`  to manually trigger a compaction action.

3. I think you should restart data nodes as well.
